gateway/webhook/webhookTransformer.go

```diff
@@ -68,7 +70,14 @@ func (bt *batchWebhookTransformerT) transform(events [][]byte, sourceType string
 		err := errors.New(response.GetStatus(response.RequestBodyReadFailed))
 		return transformerBatchResponseT{batchError: err, statusCode: statusCode}
 	}
-
+	/*
+		Following 404 case can be in that case when v1 endpoint is not there
+		that is new server is hitting old transformer( when there used to be no source sent to tranformer)
+	*/
+	if version == "v1" && resp.StatusCode == 404 {
+		// v1 endpoint not available so falling back to v0 endpoint
+		return bt.transform(payloadArr, events, sourceType, "v0")
+	}
```

Comment:
dont use recursion here check if 404 was recievd then make another call

```
transformStart := time.Now()
resp, err := bt.webhook.netClient.Post(url, "application/json; charset=utf-8", bytes.NewBuffer(payload))
bt.stats.transformTimerStat.Since(transformStart)

```
make this into a small function here and call it again incase of 404

Reply:
Recursion removed .
And a different function is made